src/ui/DynamicFacetSearch/DynamicFacetSearchValueRenderer.ts

```diff
@@ -21,16 +21,15 @@ export class DynamicFacetSearchValueRenderer implements IValueRenderer {
 
   private renderCheckbox() {
     this.valueCheckbox = new DynamicFacetValueCheckbox(this.facetValue);
-    $$(this.valueCheckbox.element)
-      .find('button')
-      .setAttribute('tabindex', '-1');
+    $$(this.valueCheckbox.element).find('button').setAttribute('tabindex', '-1');
 
     this.dom.append(this.valueCheckbox.element);
   }
 
   public selectAction() {
     this.facet.enableFreezeFacetOrderFlag();
     this.facet.toggleSelectValue(this.facetValue.value);
+    this.facetValue.select();
```

Comment:
So basically, we pass the string value to the facet to cause it to select the value but facetValue is never updated to selected. The state is used inside `logSelectActionToAnalytics` to determine whether to send the select or deselect method. I'm not sure if there's a better way to this that doesn't involve forcing the state to selected here however.

Reply:
Why isn't the state correctly updated on `this.facetValue` after having been passed through `this.facet.toggleSelectValue(this.facetValue.value);`, though ?

Reply:
In any case, the side effect of `facetValue.select()` are relatively limited: it just update the state. 

So it seems pretty unlikely that this would cause other bugs.

Reply:
They `toggleSelect` inside `this.facet.toggleSelectValue(this.facetValue.value)` points at the facetValue stored inside the facet, this is different than the the facet value in the `selectAction` method which points to the facet search result value. So it is working, but its toggling the value inside the facet, not the facet search.

Reply:
Yeah it looks like it won't cause any unexpected issues 😄